src/main/java/org/spongepowered/common/service/pagination/SpongePaginationList.java

```diff
@@ -46,19 +45,23 @@
 import java.util.stream.Collectors;
 import java.util.stream.StreamSupport;
 
+import javax.annotation.Nullable;
+
 @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
-public class SpongePaginationList implements PaginationList {
+public final class SpongePaginationList implements PaginationList {
 
     private final SpongePaginationService service;
-    private final Iterable<Text> contents;
-    private final Optional<Text> title;
-    private final Optional<Text> header;
-    private final Optional<Text> footer;
-    private final Text paginationSpacer;
+    private final Iterable<Component> contents;
+    private final Optional<Component> title;
```

Comment:
More of a note for the future - but storing an optional just feels _wrong_. It's being used to avoid optional (re)creation in getters, but... how often are the getters called? Not often I'm sure! The `sendTo` method actually unwraps the optional into a null.

Perhaps this is a good time to ditch the optionals and just have nullable fields.